src/utils/post_utils.ts

```diff
@@ -25,6 +25,10 @@ export function isSystemMessage(post: Post): boolean {
     return Boolean(post.type && post.type.startsWith(Posts.SYSTEM_MESSAGE_PREFIX));
 }
 
+export function isSystemMessageAddToChannel(post: Post, userId: $ID<UserProfile>): boolean {
```

Comment:
I think this generally works, but what would you think about combining this with `shouldIgnorePost` instead of having two separate functions? If both of the functions are only ever used together, than keeping them together would likely make sense.

Reply:
I considered that, I just wasn't sure if it was worth adding complexity the `shouldIgnorePost` function if we may remove it from redux. `isSystemMessage` and `shouldIgnorePost` are already very similar and it feels like they could be merged.

Reply:
They are slightly different in that not all types of system messages are ignored. `shouldIgnorePost` could get moved from redux to the web app, but I doubt we'll outright remove it. Overall, I added `shouldIgnorePost` to encapsulate all the conditions for ignoring a post (aka being a join/leave message), but I gave it the generic name so that its purpose would be very clear.

Reply:
That approach make sense - I updated the function to include this case.